crates/rpc/rpc/src/eth/helpers/transaction.rs

```diff
@@ -41,6 +45,43 @@ where
 
         Ok(hash)
     }
+
+    async fn send_raw_transaction_sync(
+        &self,
+        tx: Bytes,
+    ) -> Result<Option<RpcReceipt<Self::NetworkTypes>>, Self::Error> {
+        let hash = self.send_raw_transaction(tx).await?;
+        let mut stream = self.provider().canonical_state_stream();
+        let timeout_sleep = tokio::time::sleep(tokio::time::Duration::from_secs(30));
+        tokio::pin!(timeout_sleep);
+
+        loop {
+            tokio::select! {
```

Comment:
I believe we can make this a bit nicer if we use the builtin tokio::time::timeout function which is what we want here